app/api/adapters/db/__init__.py

```diff
@@ -0,0 +1,89 @@
+"""
+Database adapter for Postgres database.
+This module is an abstraction around the psycopg3 python postgres library. It
+has methods for connecting to the database, and abstracts away the loading of
+config and environment variables.
+
+Usage:
+    # During application initialization
+    from api.adapters import db
+    db.init()
+
+    # In a route handler
+    from api.adapters import db
+    with db.get_connection() as conn:
+        with conn.cursor() as cur:
+            cur.execute("SELECT 1")
+"""
+
+import atexit
+import os
+
+import psycopg
+import psycopg_pool
+from psycopg import rows
+
+from api import logging
+
+# Re-export these types to restrict the API of this module
+# to only the types we want to expose explicitly.
+Connection = psycopg.Connection
+Cursor = psycopg.Cursor
+DBConnectionError = psycopg.OperationalError
+kwargs_row = rows.kwargs_row
+
+logger = logging.get_logger(__name__)
+
+
+def init():
+    """Initializes the database connection pool"""
+    global pool
+    print(get_conninfo())
+    pool = psycopg_pool.ConnectionPool(get_conninfo(), open=False)
+
+    # Open the connection pool when the app starts
+    pool.open()
+
+    # # Close the connection pool when the app shuts down
+    atexit.register(pool.close)
+
+
+def get_conninfo() -> str:
+    """Gets a db connection info based on environment variables"""
+    dbhost = os.getenv("DB_HOST")
+    dbname = os.getenv("POSTGRES_DB")
+    username = os.getenv("POSTGRES_USER")
+    password = os.getenv("POSTGRES_PASSWORD")
+    ENVIRONMENT = os.getenv("ENVIRONMENT", "production")
+    sslmode = "allow" if ENVIRONMENT == "local" else "verify-full"
+
+    return psycopg.conninfo.make_conninfo(
+        host=dbhost,
+        dbname=dbname,
+        user=username,
+        password=password,
+        sslmode=sslmode,
+    )
+
+
+def get_connection() -> psycopg.Connection:
+    """Gets a db connection
+    Uses database credentials in environment variables to obtain a database
+    connection.
+    Usage:
+      with get_connection() as conn:
+        with conn.cursor() as cur:
+          cur.execute("SELECT 1")
+          result = cur.fetchall()
```

Comment:
The security linting tool that runs (Bandit) isn't going to like this, it doesn't let you write raw-sql with any params passed into the text. Will either need to replace it, or ignore all of those warnings which are in place to prevent any sort of SQL injection. Might want some sort of util for making SQL.

I'd also push back on any approach that doesn't come with at least some guidance on how to deserialize a response into something we can work with, the primary benefit of SQLAlchemy in my mind. It's one thing to do `select id from my_table` another to do `select a, b, c, d, e, f, g ... from my_table`.